lib/seahorn/Smt/TypeChecker.cc

```diff
@@ -12,64 +12,72 @@ namespace {
 
 //==-- main implementation goes here --==//
 class TCVR {
-  ExprVector expTypeStack;
-
-  bool isConst (Expr exp) {
-   return bind::isBoolConst(exp) || bind::isIntConst(exp);
+  ExprMap exprMap;
+
+  bool isBool(Expr exp) {
```

Comment:
this is good in principle, but you want the code to be parametric in types. So you want it to return the type, rather than assume that you know the type names